src/resources/listings.js

```diff
@@ -99,6 +108,63 @@ class Listings extends ResourceBase {
     return range(0, Number(listingsLength))
   }
 
+  async search(query) {
+    // Issues an API call to the bridge server for searching listings.
+    // Args:
+    //   query(SearchQuery): query object.
+    // Returns:
+    //   list(int): listing ids that match the query.
+    if (query.rawQuery == '') {
+      return []
+    }
+
+    // Make the search API call to the bridge server.
+    const url = encodeURI(
+      appendSlash(this.indexingServerUrl) + 'api/search/listings?query=' + query.rawQuery)
+    const response = await this.fetch(url, { method: 'GET' })
+    if (response.status != 200) {
+      throw new Error('Search API call failed with status ', response.status)
+    }
+    const json = await response.json()
+
+    // Extract listing addresses from the "_id" field of the search results.
+    const addresses = json.listings.map(listing => listing._id)
+
+    // TODO(franck): Fix once we move to new contracts.
+    // Currently the React ListingGrids component manipulates listing ids.
+    // But the search API only returns addresses so we need to convert those into ids.
+    // There is no method on the listing registry contract to handle this conversion.
+    // As a workaround we build a lookup table of address -> id.
+    // While fine in a development environment with a few listings, it won't scale in production
+    // since it involves calling listingsRegistryContract as many times as there are listings...
+    if (process.env.NODE_ENV === 'production') {
+      throw new Error('Search not supported yet in production.')
+    }
+
+    const listingIds = await this.allIds()
+
+    // Get the listing registry contract deployed on the blockchain.
+    const contract = this.contractService.listingsRegistryContract
+    const listingsRegistry = await this.contractService.deployed(contract)
+
+    // Build a map of address -> id.
+    const addressToId = {}
+    await Promise.all(
+      listingIds.map(async id => {
```

Comment:
This seems odd to me. The `map` method is used to return an array of mapped values. If you just want to iterate without returning anything, `forEach` is all you need.

That said I much prefer to use `map` and not actually modify things in place. I think there's a way to do this here with some refactoring. I'll come back to this.